crates/polars-plan/src/dsl/function_expr/pow.rs

```diff
@@ -24,15 +24,52 @@ impl Display for PowFunction {
     }
 }
 
-fn pow_on_floats<T>(base: &ChunkedArray<T>, exponent: &Series) -> PolarsResult<Option<Series>>
+fn pow_on_chunked_arrays<T, F>(
+    base: &ChunkedArray<T>,
+    exponent: &ChunkedArray<F>,
+) -> PolarsResult<Option<Series>>
+where
+    T: PolarsNumericType,
+    F: PolarsNumericType,
+    T::Native: num::pow::Pow<F::Native, Output = T::Native> + ToPrimitive,
+    ChunkedArray<T>: IntoSeries,
+{
```

Comment:
this is mostly just factored out from the old `pow_on_floats` so it can be shared between the new `pow_on_floats` and the new `pow_on_integral_dtypes`